cmd/operator-sdk/main.go

```diff
@@ -81,3 +84,70 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func checkDepManagerForCmd(cmd *cobra.Command) (err error) {
+	// Do not perform this check if wd is not in the project root,
+	// as some sub-commands might not require project root.
+	if err := projutil.CheckProjectRoot(); err != nil {
+		return nil
+	}
+
+	var dm projutil.DepManagerType
+	switch n := cmd.Name(); n {
+	case "new", "migrate":
+		// Do not perform this check if the new project is non-Go, as they do not
+		// have (Go) dep managers.
+		if n == "new" {
+			projType, err := cmd.Flags().GetString("type")
+			if err != nil {
+				return err
+			}
+			if projType != "go" {
+				return nil
+			}
+		}
+		// "new" and "migrate" commands are for projects that establish which
+		// dep manager to use. "new" should not be called if we're in the project
+		// root but could be, so check it anyway.
+		dmStr, err := cmd.Flags().GetString("dep-manager")
+		if err != nil {
+			return err
+		}
+		dm = projutil.DepManagerType(dmStr)
+	default:
+		// version and completion commands are able to be run anywhere.
+		if n == "version" || n == "completion" {
+			return nil
+		}
+		// Do not perform this check if the project is non-Go, as they do not
+		// have (Go) dep managers.
+		if !projutil.IsOperatorGo() {
+			return nil
+		}
+		if dm, err = projutil.GetDepManagerType(); err != nil {
+			return err
+		}
+	}
+
+	switch dm {
+	case projutil.DepManagerGoMod:
+		goModOn, err := projutil.GoModOn()
+		if err != nil {
+			return err
+		}
+		if !goModOn {
+			return fmt.Errorf(`depedency manger "modules" requires wd be in $GOPATH/src` +
+				` and GO111MODULE=on, or outside of $GOPATH/src and GO111MODULE="on", "auto", or unset`)
```

Comment:
I ran into the following error trying this locally with Go modules turned off:
```console
$ operator-sdk new app-operator
INFO[0000] Creating new Go operator 'app-operator'.
FATA[0000] Dependency manager "modules" has been selected but go modules are not active. Activate modules then run "operator-sdk new app-operator".
```
That's from here:
https://github.com/operator-framework/operator-sdk/blob/master/cmd/operator-sdk/new/cmd.go#L185-L186

It's not a part of this PR, but can we change that error message to be the same as this one. So users know how to activate Go modules.